src/amo/components/UserProfileEdit/index.js

```diff
@@ -0,0 +1,364 @@
+
+              {/*
+                TODO: Don't show these to users who don't have a public-facing
+                user profile page (eg are developers). It's just noise and may
+                encourage them to enter a lot of text (especially the bio) which
+                no one will see. It also gets in the way of settings,
+                like notifications, below.
+              */}
+              <label className="UserProfileEdit--label" htmlFor="homepage">
+                {i18n.gettext('Homepage')}
+              </label>
+              <input
+                className="UserProfileEdit-homepage"
+                disabled={!user}
+                id="homepage"
+                name="homepage"
+                ref={(ref) => { this.homepage = ref; }}
+                defaultValue={user && user.homepage}
+                type="url"
+              />
+              <p className="UserProfileEdit--help">
+                {i18n.gettext(`This URL will only be visible if you are a
+                  developer.`)}
+              </p>
+
+              <label className="UserProfileEdit--label" htmlFor="location">
+                {i18n.gettext('Location')}
+              </label>
+              <input
+                className="UserProfileEdit-location"
+                disabled={!user}
+                id="location"
+                name="location"
+                ref={(ref) => { this.location = ref; }}
+                defaultValue={user && user.location}
+              />
+
+              <label className="UserProfileEdit--label" htmlFor="occupation">
+                {i18n.gettext('Occupation')}
+              </label>
+              <input
+                className="UserProfileEdit-occupation"
+                disabled={!user}
+                id="occupation"
+                name="occupation"
+                ref={(ref) => { this.occupation = ref; }}
+                defaultValue={user && user.occupation}
+              />
+            </Card>
+
+            <Card
+              className="UserProfileEdit--Card"
+              header={i18n.gettext('Your Biography')}
+            >
+              <label className="UserProfileEdit--label" htmlFor="biography">
+                {i18n.gettext(`Introduce yourself to the community. This text
+                  will appear on your user profile page.
+                  (Some HTML supported)`)}
+              </label>
+              <Textarea
+                className="UserProfileEdit-biography"
+                disabled={!user}
+                id="biography"
+                name="biography"
+                ref={(ref) => { this.biography = ref; }}
+                defaultValue={user && user.biography}
+              />
+            </Card>
+
+            <div className="UserProfileEdit-buttons-wrapper">
+              <Button
+                buttonType="action"
+                className="UserProfileEdit-submit-button UserProfileEdit-button"
+                disabled={!user || isEditing}
```

Comment:
As we've done with the edit collection form, perhaps we should set the button to disabled if the form is invalid. Specifically, if username is empty.

Reply:
Indeed, and this is a good argument in favor of controlled inputs!